code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -148,10 +148,9 @@ func (sp *stakePool) addOffer(amount state.Balance) {
 	sp.TotalOffers += amount
 }
 
-// extendOffer changes offer lock and expiration on update allocations
-func (sp *stakePool) extendOffer(delta state.Balance) (err error) {
-	sp.TotalOffers += delta
-	return
+// remove offer of an allocation related to blobber owns this stake pool
+func (sp *stakePool) removeOffer(amount state.Balance) {
```

Comment:
We'd better explicit check if the amount is > 0, otherwise it will be used as `addOffer`, so as the `addOffer`.